framework/include/problems/FEProblemBase.h

```diff
@@ -1842,10 +1842,37 @@ class FEProblemBase : public SubProblem, public Restartable
   }
 
   /**
-   * Whether or not the invalid solutions are allowed
+   * Whether or not has solution warning flagged
+   */
+  bool hasSolutionWarning() { return _has_solution_warning; }
+
+  bool hasSolutionWarning(bool state) { return _has_solution_warning = state; }
+
+  /**
+   * Whether or not has invalid solution flagged
+   */
+  bool hasInvalidSolution() { return _has_invalid_solution; }
+
+  bool hasInvalidSolution(bool state) { return _has_invalid_solution = state; }
+
+  /**
+   * Whether or not to allow convergence of the solution
```

Comment:
it s more "accept the solution" than allow convergence.
Convergence is residuals going to 0